internal/db/schema/migrations/oss/postgres/80/10_auth_method_base_table_updates.up.sql

```diff
@@ -0,0 +1,138 @@
+  as $$
+  begin
+    update auth_method set update_time = now() where public_id = new.public_id;
+    return new;
+  end;
+  $$ language plpgsql;
+  comment on function update_auth_method_table_update_time() is
+    'update_auth_method_table_update_time is used to automatically update the update_time '
+    'of the base table whenever one of the subtype tables are updated';
+
+  -- Add trigger to update the new is_active_public_state column on ldap and oidc subtype update.
+  -- Password subtype has no state and will always be considered active public.
+  create function update_auth_method_table_is_active_public_state() returns trigger
+  as $$
+  begin
+    if new.state = 'active-public' then
+      update auth_method set is_active_public_state = true where public_id = new.public_id;
+      return new;
+    else
+      update auth_method set is_active_public_state = false where public_id = new.public_id;
+      return new;
+    end if;
+  end;
+  $$ language plpgsql;
+  comment on function update_auth_method_table_is_active_public_state() is
+    'update_auth_method_table_is_active_public_state is used to automatically update the '
+    'is_active_public_state column of the base table whenever one of the subtype tables are updated';
+
+  -- Add triggers to subtype tables
+  create trigger update_auth_method_table_update_time before update on auth_ldap_method
+    for each row execute procedure update_auth_method_table_update_time();
+  create trigger update_auth_method_table_update_time before update on auth_oidc_method
+    for each row execute procedure update_auth_method_table_update_time();
+  create trigger update_auth_method_table_update_time before update on auth_password_method
+    for each row execute procedure update_auth_method_table_update_time();
+
+  -- Replace trigger on password subtype table
+  drop trigger insert_auth_method_subtype on auth_password_method;
+  create trigger insert_auth_method_password_subtype before insert on auth_password_method
+    for each row execute procedure insert_auth_method_password_subtype();
+
+  -- Add is_active_public_state triggers to ldap and oidc tables, as password has no state column.
+  create trigger update_auth_method_table_is_active_public_state before update on auth_ldap_method
+    for each row execute procedure update_auth_method_table_is_active_public_state();
+  create trigger update_auth_method_table_is_active_public_state before update on auth_oidc_method
+    for each row execute procedure update_auth_method_table_is_active_public_state();
+
+  -- Add new indexes for the create and update time queries.
+  create index auth_method_create_time_public_id_idx
+      on auth_method (create_time desc, public_id asc);
+  create index auth_method_update_time_public_id_idx
+      on auth_method (update_time desc, public_id asc);
+
+  -- Make column not null
+  alter table auth_method alter column is_active_public_state set not null;
```

Comment:
Minor nit but maybe move this to just after the data migrations, and also make the comment less generic 😄?